tools-local/tasks/mobile.tasks/WasmBundleTask/WasmBundleTask.cs

```diff
@@ -0,0 +1,158 @@
+using Microsoft.Build.Utilities;
+using System.Diagnostics;
+
+public class WasmBundleTask : Task
+{
+    public string? InputDirectory { get; set; }
+    public string? OutputDirectory { get; set; }
+    public string? Type { get; set; }
+    public string? FileName { get; set; } 
+
+    private (byte[] json_bytes, MemoryStream stream) enumerateData (string[] sub_folders, string output_folder) {
+        var indices = new List<object[]>();
+        var stream = new MemoryStream();
+
+        foreach (var folder in sub_folders)
+        {
+            var path = Path.Combine(output_folder, folder);
+            if (folder == "zone1970.tab") {
+                var fileInfo = new FileInfo(path);
+                indices.Add(new object[] { "zone.tab", fileInfo.Length});
+                string[] systemtz = { "America/Los_Angeles", "Australia/Sydney", "Europe/London", "Pacific/Tongatapu", 
+                                "America/Sao_Paulo", "Australia/Perth", "Africa/Nairobi", "Europe/Berlin",
+                                "Europe/Moscow", "Africa/Tripoli", "America/Argentina/Catamarca", "Europe/Lisbon",
+                                "America/St_Johns"};
+                using (var readStream = fileInfo.OpenRead()) 
+                using (StreamReader sr = new StreamReader(readStream)) 
+                using (MemoryStream ms = new MemoryStream())
+                using (StreamWriter ws = new StreamWriter(ms)){
+                    string? line;
+                    while ((line = sr.ReadLine()) != null) {
+                        if (systemtz.Any(x => Regex.IsMatch(line, $@"\b{x}\b"))) {
+                            ws.WriteLine(line);
+                        }
+                    }
+                    ws.Flush();
+                    ms.Position = 0;
+                    ms.CopyTo(stream);
+                }
+            } else {
+                var directoryInfo = new DirectoryInfo(path);
+                foreach (var entry in directoryInfo.EnumerateFiles("*", SearchOption.AllDirectories))
+                {
+                    var relativePath = entry.FullName.Substring(output_folder.Length).Trim('/');
+                    indices.Add(new object[] { relativePath, entry.Length});
+
+                    using (var readStream = entry.OpenRead())
+                        readStream.CopyTo(stream);
+                }
+            }
+        }
+        
+        stream.Position = 0;
+        var options = new JsonSerializerOptions
+        {
+            WriteIndented = false
+        };
+
+        var jsonBytes = JsonSerializer.SerializeToUtf8Bytes(indices);
+        
+        return (jsonBytes, stream);
+    }
+
+    private (byte[] json_bytes, MemoryStream stream) readTimeZone (string folder) {
+        // https://en.wikipedia.org/wiki/Tz_database#Area
+        var areas = new[] { "Africa", "America", "Antarctica", "Arctic", "Asia", "Atlantic", "Australia", "Europe", "Indian", "Pacific", "zone1970.tab"};
```

Comment:
Should these areas be accepted as an input by the task so that in the future is controllable from outside of it via a MSBuild item?